src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -79,65 +72,35 @@ class GitVersionPlugin implements Plugin<Project> {
-        GitCli.verifyGitCommandExists()
-
-        def runGitCmd = { String... commands ->
-            return GitCli.runGitCommand(project.projectDir, commands);
+        // This used to be implemented with JGit and replaced with shelling out to installed git (#46) because JGit
+        // didn't support required behavior. Using installed git doesn't work in some environments or
+        // with older versions of git client. We're switching back to implementation with JGit. To make sure we don't
+        // make breaking change, we're keeping both implementations. Plan is to get rid of installed git implementation.
+        // TODO(mbakovic): Use JGit only implementation #87
+
+        String nativeGitDescribe = new NativeGitDescribe(project.projectDir).describe(prefix)
+        String jgitDescribe = new JGitDescribe(project.projectDir).describe(prefix)
+
+        // If native failed, return JGit one
+        if (nativeGitDescribe == null) {
+            return jgitDescribe
         }
 
-        Git git = gitRepo(project)
-        try {
-            // back-compat: the JGit "describe" command throws an exception in repositories with no commits, so call it
-            // first to preserve this behavior in cases where this call would fail but native "git" call does not.
-            new DescribeCommand(git.getRepository()).call()
-
-            /*
-             * Mimick 'git describe --tags --always --first-parent --match=${prefix}*' by using rev-list to
-             * support versions of git < 1.8.4
-             */
-
-            // Get SHAs of all tags, we only need to search for these later on
-            Set<String> tagRefs = Sets.newHashSet()
-            for (String tag : getLines(runGitCmd("show-ref", "--tags", "-d"))) {
-                List<String> parts = WORD_SPLITTER.splitToList(tag)
-                Preconditions.checkArgument(parts.size() == 2, "Could not parse output of `git show-ref`: %s", parts)
-                tagRefs.add(parts.get(0))
-            }
-
-            List<String> revs = getLines(runGitCmd("rev-list", "--first-parent", "HEAD"))
-            for (int depth = 0; depth < revs.size(); depth++) {
-                String rev = revs.get(depth)
-                if (tagRefs.contains(rev)) {
-                    String exactTag = runGitCmd("describe", "--tags", "--exact-match", "--match=${prefix}*", rev)
-                    if (exactTag != "") {
-                        return depth == 0 ?
-                                exactTag : String.format("%s-%s-g%s", exactTag, depth, abbrevHash(revs.get(0)))
-                    }
-                }
-            }
-
-            // No tags found, so return commit hash of HEAD
-            return abbrevHash(runGitCmd("rev-parse", "HEAD"))
-        } catch (Throwable t) {
-            return null
+        // If native succeeded, make sure it's same as JGit one
+        if (!nativeGitDescribe.equals(jgitDescribe)) {
```

Comment:
This is actually behaviour we want. If native succeeds, it must be equal with jgit. So if jgit fails and native succeeds we'll raise an exception. That should never happen!